examples/jvm-interop/bridge.c

```diff
@@ -0,0 +1,387 @@
+#include <errno.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <stddef.h>
+#include <string.h>
+#include <unistd.h>
+#include <stdalign.h>
+#include <stdint.h>
+#include <setjmp.h>
+
+#include <jni.h>
+#include "javaSource_Demo.h"
+
+JavaVM* vm;
+
+#define ERR_MSG_MAX_SIZE 256
+
+jmp_buf exception_buffer;
+char* err_msg[ERR_MSG_MAX_SIZE] = {0};
+
+jint JNI_OnLoad(JavaVM *loadedVM, void *reserved)
+{
+    // https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/invocation.html
+    vm = loadedVM;
+    return JNI_VERSION_1_2;
+}
+
+void *roc_alloc(size_t size, unsigned int alignment)
+{
+    return malloc(size);
+}
+
+void *roc_realloc(void *ptr, size_t new_size, size_t old_size,
+                  unsigned int alignment)
+{
+    return realloc(ptr, new_size);
+}
+
+void roc_dealloc(void *ptr, unsigned int alignment)
+{
+    free(ptr);
+}
+
+__attribute__((noreturn)) void roc_panic(void *ptr, unsigned int alignment)
```

Comment:
Just double checked on zulip, the first arg is actually a `RocStr`. So should be `struct RocStr *msg` instead of `void *ptr`. Also, the pointer should never be null, but of course it may be a small string. Oh, also, the second field is not alignment (this probably originates from copy and paste). It should be the `tag_id`. Basically, it tells the origin of the crash. Userland, roc primitive, etc. You can still probably just ignore it though.

Reply:
fixed